cirq-core/cirq/ops/measurement_gate.py

```diff
@@ -312,7 +312,18 @@ def _act_on_(self, sim_state: 'cirq.SimulationStateBase', qubits: Sequence['cirq
 
         if not isinstance(sim_state, SimulationState):
             return NotImplemented
-        sim_state.measure(qubits, self.key, self.full_invert_mask(), self.confusion_map)
+        try:
+            sim_state.measure(
+                qubits, self.key, self.full_invert_mask(), confusion_map=self.confusion_map
+            )
+        except TypeError as e:
+            # Ensure that the error was due to confusion_map.
+            if not any("unexpected keyword argument 'confusion_map'" in arg for arg in e.args):
+                raise
+            _compat._warn_or_error(
+                "Starting in v0.16, SimulationState subclasses will be required to accept "
+                "a 'confusion_map' argument. See SimulationState.measure for details."
+            )
```

Comment:
Should retry the `measure` call without the confusion map?

Reply:
Done.